### ISSUE TITLE
core: getNumThreads() result is wrong on Win32/Concurrency backend

### ISSUE THREAD

**Issue by alalek** [opencv/opencv#16777]
getNumThreads() returns `nCPUs+1` by default:
- 4 cores: `getNumThreads()` => 5
- 4 cores + HT: `getNumThreads()` => 9

relates https://github.com/opencv/opencv/commit/460644b8a4325558414f1d30ba15138857434875

**Comment by saskatchewancatch** [comment]
OpenCV doc for [getNumThreads:Concurrency](https://docs.opencv.org/3.4/db/de0/group__core__utils.html#ga2db334ec41d98da3129ef4a2342fc4d4) says "_The number of threads, that OpenCV will try to use for parallel regions._"

Offending code is below: I can't imagine why 1 would be added to whatever's returned by `GetNumberOfVirtualProcessors()`. 

```
#elif defined HAVE_CONCURRENCY

    return 1 + (pplScheduler == 0
        ? Concurrency::CurrentScheduler::Get()->GetNumberOfVirtualProcessors()
        : pplScheduler->GetNumberOfVirtualProcessors());
```

There's no reason a parallelized implementation should use more than number of vCPUs than system. I can't seem to find anything in the MS Concurrency runtime's documentation that says a task will be more optimal if a scheduler is configured to use 1 plus number of vCPUs. Interesting find.

**Comment by alalek** [comment]
`1 +` is from `setNumThreads()` implementation:

https://github.com/opencv/opencv/blob/3befdb4ae8f59bedff5ace11ea873bb404599f5a/modules/core/src/parallel.cpp#L726-L731

Probably `1` here is for the "caller" thread.

---

setNumThreads(2) + getNumThreads() should be checked too.

**Comment by saskatchewancatch** [comment]
@alalek , thanks for pointing that out. I

If I have understood correctly, for `setNumThreads`, then I propose patch below. Not sure what you mean by `getNumThreads() should be checked too` as the PR contains patch for that.

```diff
diff --git a/modules/core/src/parallel.cpp b/modules/core/src/parallel.cpp
index af520633ae..f967953643 100644
--- a/modules/core/src/parallel.cpp
+++ b/modules/core/src/parallel.cpp
@@ -701,11 +701,11 @@ void setNumThreads( int threads_ )
         // Concurrency always uses >=2 threads, so we just disable it if 1 thread is requested
         numThreads = 0;
     }
-    else if (pplScheduler == 0 || 1 + pplScheduler->GetNumberOfVirtualProcessors() != (unsigned int)threads)
+    else if (pplScheduler == 0 || pplScheduler->GetNumberOfVirtualProcessors() != (unsigned int)threads)
     {
         pplScheduler = Concurrency::Scheduler::Create(Concurrency::SchedulerPolicy(2,
-                       Concurrency::MinConcurrency, threads-1,
-                       Concurrency::MaxConcurrency, threads-1));
+                       Concurrency::MinConcurrency, threads,
+                       Concurrency::MaxConcurrency, threads));
     }

 #elif defined HAVE_PTHREADS_PF

```

Quick test with on 4 core system:
```
    cout << endl << getNumThreads();
    cv::setNumThreads(3); //scheduler is created
    cout << endl << cv::getNumThreads(); //scheduler is queried for vCPU count
```
returns:
```
4

3
```

This is good?

**Comment by alalek** [comment]
> threads-1

Check actual CPU load and number of active threads.
`threads-1` is correct, because the current thread is attached into thread pool before running job.

---

Fixed by #16779
